instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go

```diff
@@ -64,18 +64,23 @@ func (s *instrumentedS3) PutObjectWithContext(ctx aws.Context, input *s3.PutObje
 		trace.WithSpanKind(trace.SpanKindClient),
 		trace.WithAttributes(attrs...),
 	)
-	defer span.End()
 
 	output, err := s.S3API.PutObjectWithContext(ctx, input, opts...)
+	callReturnTime := trace.WithTimestamp(time.Now())
+	defer span.End(callReturnTime)
 
 	if err != nil {
 		attrs = append(attrs, labelStatusFailure)
+		span.SetAttributes(labelStatusFailure)
+		span.SetStatus(codes.Ok, err.Error())
```

Comment:
is that expected to use `codes.ok` even there is an error?

Reply:
ah that's a mistake, thanks for picking up will fix